dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -401,6 +407,30 @@ private void updateTags(List<Tag> newTags, @NotNull Tool tool, SourceCodeRepoInt
         toolDAO.create(tool);
     }
 
+    private void updateImageInformation(Tool tool, Tag newTag, Tag oldTag) {
+        // If old tag does not have image information yet, try to set it. If it does, potentially old tag could have been deleted on
+        // GitHub and replaced with tag of the same name. Check that the image is the same. If not, replace.
+        if (oldTag.getImages().isEmpty()) {
+            oldTag.getImages().addAll(getImageInformationForTag(tool, newTag).getImages());
+
+        } else {
+            // There should only be one image per tag so each loop should only execute once.
+            Iterator<Image> iterator = oldTag.getImages().iterator();
+            Iterator<Image> iterator2 = newTag.getImages().iterator();
+
+            while (iterator.hasNext()) {
+                Image oldImage = iterator.next();
+                while (iterator2.hasNext()) {
+                    Image newImage = iterator2.next();
+                    if (oldImage.getImageID() != newImage.getImageID()) {
```

Comment:
You need to use `equals` instead of `!=`. If there is a possibility of the either imageId being null (I don't think there is), then you should use `Object.equals`.